### PR TITLE
Spawn rework ports

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
## About The Pull Request
Ports
https://github.com/discordia-space/CEV-Eris/pull/5374
https://github.com/discordia-space/CEV-Eris/pull/5508
https://github.com/discordia-space/CEV-Eris/pull/5510
https://github.com/discordia-space/CEV-Eris/pull/5514
https://github.com/discordia-space/CEV-Eris/pull/5516
https://github.com/discordia-space/CEV-Eris/pull/5518
https://github.com/discordia-space/CEV-Eris/pull/5556

Basically ports over all of the rework prs and their map changes hopefully without fail.

# not tested but it complies and matches 90-95% of what Eris has
## Changelog
```changelog
tweak: All randomness of what can and cant spawn has been now updated to be the same as what Eris has done.
```

### CODE REVIEW

**Review by Michiyamenotehifunana — APPROVED** [review]
This is the port of the loot rework that I had instructed Trilby to work on. As can be seen from the 662 files changed, this is basically unreviewable. However, I believe Trilby has tested it at the very least and it compiles.

**Comment by Trilbyspaceclone** [comment]
@Michiyamenotehifunana waiting on
https://github.com/SyzygyStation/Syzygy-Eris/pull/187
before I make the map edits needed for it to all complie

**Comment by ShadzFen** [comment]
Also, I know it's a draft, but you might want to change the title to be a little bit more descriptive. Something something upstream PR or whatever.

**Comment by Trilbyspaceclone** [comment]
It will be when ready. draft pr for draft
